docs/data-imports.md

```diff
@@ -0,0 +1,52 @@
+# Data Imports
+
+The data imports pick up where the ETL leaves off by copying and importing the
+CSV files into the Postgresql database. This document describes that process in
+more detail and also some of the schema choices that were made to optimize the
+imports for speed.
+
+The examples below will use Firefox on Android as the example, as it is a Glean
+based product and will be the example for future products in GLAM going forward.
+
+## Schema
+
+The Firefox for Desktop (aka fenix) product has 2 primary tables assocated with
+its data. The aggregation table and the materialized view on top of that data.
+
+The aggregation table exist so imports can be as fast as possible. It has
+minimal indexes and constraints so data can import quickly. The only index is
+the primary key. The only constraint is a unique constraint comprised of the
+dimensions of the data (e.g. channel, build ID, metric, etc.) so that we can
+assure we have only 1 row across these dimensions. This unique constraint is
+also used in the "upsert" so that if new data comes in we insert it, or if a
+matching record exists we update the data (counts, histograms, and percentiles).
+
+The materialized view exists so the API queries are as fast as possible. It is a
+direct copy of the aggregation table but with added indexes, roughly to match
+the top level selectors in GLAM so the SQL `WHERE` clauses are fast. Since the
+materialized view has indexes, refreshing it is time and resource intensive, but
+it is performed `CONCURRENTLY` so that the API still functions while it is being
+refreshed.
+
+Firefox for Desktop is similar to the above except we break down the data
+further by channel due to the size of the data, so each channel has the above
+described two tables each.
+
+## CSV imports
+
+The import process kicks off via the Django management commands. For Firefox for
+Android the execution looks like:
+
+    python manage.py import_glean_aggs nightly
+
+The following are the steps that the imoprts follow to import the data:
+
+- **Create a list of files** to import by pattern matching all files in the GCS
+  bucket, matching for Fenix and the app ID
+- For each file:
+  - we **create a temporary table** and **load the CSV data** into this
+    temporary table using the Postgresql `COPY` command,
+  - then we **perform an "upsert"** to quickly move the data into the main
+    aggregation table
+- Once all files have been imported, we **refresh the materialized view** so the
+  new data is shared via the API
```

Comment:
If it's not too much work, could you reference the code for each step here?

Reply:
Yep, I can add that.

Reply:
I ended up not linking to each part of the file directly, but tried to align the description better with the code so it's easier to follow along. My fear is the code will change out from under the docs and make the docs more confusing.